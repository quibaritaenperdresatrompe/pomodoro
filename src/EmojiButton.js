import { func, oneOf, string } from "prop-types";
import { makeStyles } from "@material-ui/styles";
import React from "react";
import { useTheme } from "./ThemeProvider";

const useStyles = makeStyles(() => ({
  button: {
    border: "none",
    background: "none",
    cursor: "pointer",
    outline: "none",
  },
  emoji: {
    color: ({ theme }) => (theme === "light" ? "black" : "white"),
    fontSize: ({ size }) => ({ small: 16, medium: 32, large: 64 }[size] || 32),
    userSelect: "none",
  },
}));

export default function EmojiButton({ emoji, label, onClick, size }) {
  const theme = useTheme();
  const classes = useStyles({ size, theme });
  return (
    <button onClick={onClick} className={classes.button}>
      <span
        aria-hidden={label ? "false" : "true"}
        aria-label={label ? label : ""}
        className={classes.emoji}
        role="img"
      >
        {emoji}
      </span>
    </button>
  );
}

EmojiButton.propTypes = {
  emoji: string,
  label: string,
  onClick: func,
  size: oneOf(["small", "medium", "large"]),
};

EmojiButton.defaultProps = {
  emoji: null,
  label: null,
  onClick: Function.prototype,
  size: "medium",
};
