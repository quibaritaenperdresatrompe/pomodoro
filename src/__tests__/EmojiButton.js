import { render } from "@testing-library/react";
import React from "react";

import EmojiButton from "../EmojiButton";

describe(EmojiButton.name, () => {
  it("renders correctly", () => {
    const { asFragment } = render(<EmojiButton />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly with an unkonwn size", () => {
    const { asFragment } = render(<EmojiButton size="xs" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
