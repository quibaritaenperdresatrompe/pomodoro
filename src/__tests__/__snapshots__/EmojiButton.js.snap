// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmojiButton renders correctly 1`] = `
<DocumentFragment>
  <button
    class="makeStyles-button-1"
  >
    <span
      aria-hidden="true"
      aria-label=""
      class="makeStyles-emoji-2 makeStyles-emoji-3"
      role="img"
    />
  </button>
</DocumentFragment>
`;

exports[`EmojiButton renders correctly with an unkonwn size 1`] = `
<DocumentFragment>
  <button
    class="makeStyles-button-4"
  >
    <span
      aria-hidden="true"
      aria-label=""
      class="makeStyles-emoji-5 makeStyles-emoji-6"
      role="img"
    />
  </button>
</DocumentFragment>
`;
